src/hybrid/parse_hybrid.c

```diff
@@ -70,41 +72,282 @@ static int parseSearchSubquery(ArgsCursor *ac, AREQ *searchRequest, QueryError *
+        hasRadius = true;
+      }
+    } else if (!strcasecmp(current, "EPSILON")) {
+      if (hasEpsilon) {
+        QueryError_SetError(status, QUERY_EDUPPARAM, "Duplicate EPSILON parameter");
+        return REDISMODULE_ERR;
+      } else {
+        AC_Advance(ac);
+        const char *value;
+        if (AC_GetString(ac, &value, NULL, 0) != AC_OK) {
+          QueryError_SetError(status, QUERY_ESYNTAX, "Invalid EPSILON value");
+          return REDISMODULE_ERR;
+        }
+        // Add as QueryAttribute (for query node processing)
+        QueryAttribute attr = {
+          .name = VECSIM_EPSILON,
+          .namelen = strlen(VECSIM_EPSILON),
+          .value = rm_strdup(value),
+          .vallen = strlen(value)
+        };
+        pvq->attributes = array_ensure_append_1(pvq->attributes, attr);
+        hasEpsilon = true;
+      }
+    } else if (!strcasecmp(current, "YIELD_DISTANCE_AS")) {
+      if (hasYieldDistanceAs) {
+        QueryError_SetError(status, QUERY_EDUPPARAM, "Duplicate YIELD_DISTANCE_AS parameter");
+        return REDISMODULE_ERR;
+      } else {
+        AC_Advance(ac);
+        const char *value;
+        if (AC_GetString(ac, &value, NULL, 0) != AC_OK) {
+          QueryError_SetError(status, QUERY_ESYNTAX, "Missing distance field name");
+          return REDISMODULE_ERR;
+        }
+
+        // As QueryAttribute (for query node processing)
+        QueryAttribute attr = {
+          .name = YIELD_DISTANCE_ATTR,
+          .namelen = strlen(YIELD_DISTANCE_ATTR),
+          .value = rm_strdup(value),
+          .vallen = strlen(value)
+        };
+        pvq->attributes = array_ensure_append_1(pvq->attributes, attr);
+        hasYieldDistanceAs = true;
+      }
+    } else {
+      QueryError_SetWithUserDataFmt(status, QUERY_EPARSEARGS, "Unknown parameter", " `%s` in RANGE", current);
+      return REDISMODULE_ERR;
+    }
+  }
+  if (!hasRadius) {
+    QueryError_SetError(status, QUERY_EPARSEARGS, "Missing RADIUS parameter");
+    return REDISMODULE_ERR;
+  }
+  return REDISMODULE_OK;
+}
+static int parseFilterClause(ArgsCursor *ac, AREQ *vectorRequest, QueryError *status) {
+  // FILTER is in our scope, advance and process it
+  AC_Advance(ac);
+  vectorRequest->query = AC_GetStringNC(ac, NULL);
+  return REDISMODULE_OK;
+}
+
 static int parseVectorSubquery(ArgsCursor *ac, AREQ *vectorRequest, QueryError *status) {
-  // Check if VSIM parameter is present
-  if (!AC_AdvanceIfMatch(ac, "VSIM")) {
+  const char *current;
+  if (AC_GetString(ac, &current, NULL, AC_F_NOADVANCE) != AC_OK || strcasecmp("VSIM", current)) {
     QueryError_SetError(status, QUERY_ESYNTAX, "VSIM parameter is required");
     return REDISMODULE_ERR;
   }
+  AC_Advance(ac);
 
-  // Advance the cursor until we encounter one of the specified keywords or reach the end
-  while (!AC_IsAtEnd(ac)) {
-    const char *cur;
-    if (AC_GetString(ac, &cur, NULL, AC_F_NOADVANCE) != AC_OK) {
-      break;
-    }
+  // Allocate ParsedVectorQuery
```

Comment:
Shouldn't we initialize the aggregation plan?

```suggestion
  // Initialize aggregation plan for vector request
  AGPLN_Init(AREQ_AGGPlan(vectorRequest));

  // Allocate ParsedVectorQuery
```

Reply:
Done